lib/src/RealIpResolver.cc

```diff
@@ -0,0 +1,201 @@
+/**
+ *
+ *  @file RealIpResolver.cc
+ *  @author Nitromelon
+ *
+ *  Copyright 2022, Nitromelon. All rights reserved.
+ *  https://github.com/drogonframework/drogon
+ *  Use of this source code is governed by a MIT license
+ *  that can be found in the License file.
+ *
+ *  Drogon
+ *
+ */
+
+#include <drogon/drogon.h>
+#include <trantor/utils/Logger.h>
+#include <drogon/plugins/RealIpResolver.h>
+
+using namespace drogon;
+using namespace drogon::plugin;
+
+struct XForwardedForParser
+{
+    explicit XForwardedForParser(std::string value)
+        : value_(std::move(value)), start_(value_.c_str()), len_(value_.size())
+    {
+    }
+
```

Comment:
Let's inherent `trantor::NonCopyable` class so we don't accidentally implicit copy this struct. But then we crash due to the `start_` pointer may not be valid after copy.

Reply:
Done